lib/findJsModulesFor.js

```diff
@@ -22,71 +19,69 @@ function findImportsFromEnvironment(
     }));
 }
 
-function findImportsFromPackageJson(
-  config: Configuration,
-  variableName: string
-): Array<JsModule> {
-  const formattedVarName = formattedToRegex(variableName);
-
-  const ignorePrefixes = config.get('ignorePackagePrefixes')
-    .map((prefix: string): string => escapeRegExp(prefix));
-
-  const depRegex = RegExp(
-    `^(?:${ignorePrefixes.join('|')})?${formattedVarName}$`
-  );
+const PACKAGE_NAME_PATTERN = /\.\/node_modules\/([^/]+)\//;
 
-  return config.get('packageDependencies')
-    .filter((dep: string): boolean => depRegex.test(dep))
-    .map((dep: string): ?JsModule => (
-      JsModule.construct({
-        relativeFilePath: `node_modules/${dep}/package.json`,
-        stripFileExtensions: [],
-        variableName,
-        workingDirectory: config.workingDirectory,
-      })
-    ))
-    .filter(Boolean);
-}
-
-function findImportsFromLocalFiles(
+function findJsModulesFromModuleFinder(
   config: Configuration,
   variableName: string,
+  finder: ModuleFinder,
   pathToCurrentFile: string
 ): Promise<Array<JsModule>> {
   return new Promise((resolve: Function, reject: Function) => {
-    findMatchingFiles(variableName, config.workingDirectory)
-    .then((files: Array<string>) => {
-      const matchedModules = [];
-      const excludes = config.get('excludes');
-      files.forEach((f: string) => {
-        const isExcluded = excludes
-          .some((globPattern: string): boolean => minimatch(f, globPattern));
-        if (isExcluded) {
-          return;
-        }
+    let isWantedPackageDependency = Boolean;
+    if (!config.get('importDevDependencies')) {
+      const packageDependencies = config.get('packageDependencies');
+      isWantedPackageDependency = (packageName: string): boolean =>
+        packageDependencies.indexOf(packageName) !== -1;
```

Comment:
We should make `config.get('packageDependencies')` return a `Set` to avoid this array scan.